.github/workflows/build.yml

```diff
@@ -218,32 +223,39 @@ jobs:
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
 
-      # Sign container
+      # Sign container via github OIDC
       - uses: sigstore/cosign-installer@v3.4.0
         if: github.event_name != 'pull_request'
 
-      - name: Sign container image
+      - name: Sign the images
```

Comment:
Without the public and private keys, how will we set the OS configuration to allow images from this repo?

Reply:
Right now, we have configured the ublue certs as trusted in all images, but now we're not using those certs, how will it work?

Reply:
So i might be a bit out the loop on the entire process, but I thought we could now use our github to validate via cosign

```
cosign verify --certificate-oidc-issuer "https://token.actions.githubusercontent.com" --certificate-identity-regexp https://github.com/ublue-os/bluefin <TAG>
```

which will allow for the image to be verified against our OIDC signature

Reply:
recently got my bluebuild image to sign with OIDC and the problem stems from that now the image is no longer signed with the same key, yet the `policy.json` still expects it to, so updating fails:
```
user@localhost ~ [1]> rpm-ostree update
Pulling manifest: ostree-image-signed:docker://ghcr.io/gerblesh/uhypr:latest
error: Creating importer: Failed to invoke skopeo proxy method OpenImage: remote error: cryptographic signature verification failed: invalid signature when validating ASN.1 encoded signature
```
we could try creating special tags and systemd units for rebasing/fixing the `policy.json` but I don't know what's the best way forward for this, and it would probably be ideal to keep workarounds like the ones I just mentioned to a minimum

Reply:
I mentioned this in Discord, but appropriate to ask here... 

What benefit(s) does signing with OIDC provide bluefin (or any ublue-os image)?

Unless I'm missing something, to trust the OIDC signed image, the policy,json still needs to provide some information to know how to validate the signatures... so one still can't do a direct rebase from upstream Silverblue to `ostree-image-signed:docker://ghcr.io/example-org/oidc-signed-image:latest` .

And it seems your example here where skopeo fails demonstrates this problem.

Reply:
removing the configuration for the individual image allows a rebase. Signing with OIDC keyless means the signature changes every time and is verfied using github's provider. The default policy.json will work on rebase if you don't override the signature verification with pubkey auth

Reply:
cosign's docs for keyless signing are here: https://docs.sigstore.dev/signing/overview/

Reply:
https://github.com/containers/image/blob/main/docs/containers-policy.json.5.md#sigstoresigned explains how to specify OIDC verification in policy.json. There's also an option for `subjectEmail`, which will ensure the image is signed by the right account and not a random GitHub account.